esp-hal/src/lcd_cam/cam.rs

```diff
@@ -135,18 +136,36 @@ pub struct Camera<'d, CH: DmaChannel = AnyDmaChannel> {
     bus_width: usize,
 }
 
-impl<'d, CH: DmaChannel> Camera<'d, CH>
-where
-    CH::P: LcdCamPeripheral,
-{
+impl<'d> Camera<'d> {
     /// Creates a new `Camera` instance with DMA support.
-    pub fn new<P: RxPins>(
+    pub fn new<P: RxPins, CH>(
+        cam: Cam<'d>,
+        channel: ChannelRx<'d, CH>,
+        descriptors: &'static mut [DmaDescriptor],
+        pins: P,
+        frequency: HertzU32,
+    ) -> Self
+    where
+        CH: PeripheralDmaChannel,
+        CH::P: LcdCamPeripheral,
+    {
+        Self::new_typed(cam, channel.degrade(), descriptors, pins, frequency)
+    }
+}
+
+impl<'d, CH: DmaChannel> Camera<'d, CH> {
+    /// Creates a new `Camera` instance with DMA support.
+    pub fn new_typed<P: RxPins>(
```

Comment:
`.degrade()` is so easy for users to call, why go through the effort of having a separate `_new_typed`?

Other places in the codebase where `_typed` methods are provided, they helped with inference, especially with timers.